generator/routing_index_generator.cpp

```diff
@@ -376,43 +430,63 @@ bool BuildRoutingIndex(string const & filename, string const & country,
+  auto const startPos = writer.Pos();
+  CrossMwmConnectorSerializer::Serialize(transitions, connectors, codingParams, writer);
+  auto const sectionSize = writer.Pos() - startPos;
 
-  map<uint32_t, osm::Id> featureIdToOsmId;
-  if (!ParseFeatureIdToOsmIdMapping(osmToFeatureFile, featureIdToOsmId))
-    return false;
+  LOG(LINFO, ("Cross mwm section generated, size:", sectionSize, "bytes"));
+}
 
-  CrossMwmConnectorPerVehicleType connectors;
+void BuildRoutingCrossMwmSection(string const & path, string const & mwmFile,
+                                 string const & country,
+                                 CountryParentNameGetterFn const & countryParentNameGetterFn,
+                                 string const & osmToFeatureFile, bool disableCrossMwmProgress)
+{
+  LOG(LINFO, ("Building cross mwm section for", country));
+  using CrossMwmId = connector::OsmId;
+  CrossMwmConnectorPerVehicleType<CrossMwmId> connectors;
+  vector<CrossMwmConnectorSerializer::Transition<CrossMwmId>> transitions;
 
-  vector<CrossMwmConnectorSerializer::Transition> transitions;
-  CalcCrossMwmTransitions(path, mwmFile, country, countryParentNameGetterFn, featureIdToOsmId,
-                          transitions, connectors);
-
-  for (size_t i = 0; i < connectors.size(); ++i)
-  {
-    VehicleType const vehicleType = static_cast<VehicleType>(i);
-    CrossMwmConnector const & connector = connectors[i];
-    LOG(LINFO, (vehicleType, "model: enters:", connector.GetEnters().size(), ", exits:",
-                connector.GetExits().size()));
-  }
+  CalcCrossMwmConnectors(path, mwmFile, country, countryParentNameGetterFn, osmToFeatureFile,
+                         transitions, connectors);
 
   // We use leaps for cars only. To use leaps for other vehicle types add weights generation
   // here and change WorldGraph mode selection rule in IndexRouter::CalculateSubroute.
   FillWeights(path, mwmFile, country, countryParentNameGetterFn, disableCrossMwmProgress,
               connectors[static_cast<size_t>(VehicleType::Car)]);
 
-  serial::CodingParams const codingParams = LoadCodingParams(mwmFile);
-  FilesContainerW cont(mwmFile, FileWriter::OP_WRITE_EXISTING);
-  FileWriter writer = cont.GetWriter(CROSS_MWM_FILE_TAG);
-  auto const startPos = writer.Pos();
-  CrossMwmConnectorSerializer::Serialize(transitions, connectors, codingParams, writer);
-  auto const sectionSize = writer.Pos() - startPos;
+  CHECK(connectors[static_cast<size_t>(VehicleType::Transit)].IsEmpty(), ());
+  SerializeCrossMwm(mwmFile, CROSS_MWM_FILE_TAG, connectors, transitions);
+}
 
-  LOG(LINFO, ("Cross mwm section generated, size:", sectionSize, "bytes"));
-  return true;
+void BuildTransitCrossMwmSection(string const & path, string const & mwmFile,
+                                 string const & country,
+                                 CountryParentNameGetterFn const & countryParentNameGetterFn,
+                                 bool disableCrossMwmProgress)
```

Comment:
You don't use disableCorssMwmProgress here, right? Do we really need it?

Reply:
Oh! Thanks. Removed.